### PR TITLE
DocApi: Introduce includeHidden option for GET on records

### PR DESCRIPTION
**Pull request by fflorent**
## Context

For issue #416, I want to fetch all the table data, including hidden fields of records, such as:
 - `manualSort`
 - `gristHelper_DisplayX`

So when putting records in the synchronized table, we keep the display fields and the sorting.

## Technical considerations

Not much stuffs, except that I change the signature of `getTableRecords` a bit to move the `optTableId` argument to an option parameter along with the new `includeHidden`.

### CODE REVIEW

**Comment by alexmojaki** [comment]
Query parameters are generally (always) lowercase, ideally one word, maybe with underscores. I suggest calling the param `hidden`, and updating `/columns` as well before merging https://github.com/gristlabs/grist-help/pull/256. @paulfitz WDYT?

File: app/server/lib/DocApi.ts [hunk 1]
```diff
@@ -201,16 +201,23 @@ export class DocWorkerApi {
...
-          ["id", "manualSort"].includes(k)
-          || k.startsWith("gristHelper_")
-        ));
+      const columnData = await getTableData(activeDoc, req, opts?.optTableId);
+      const fieldNames = Object.keys(columnData).filter((k) => {
+        if (k === "id") {
+          return false;
+        }
+        if (
+          !isAffirmative(opts?.includeHidden) &&
```
Comment (alexmojaki) [inline app/server/lib/DocApi.ts:212]:
```suggestion
          !opts?.includeHidden &&
```

Comment (fflorent) [reply app/server/lib/DocApi.ts:212]: Thanks!

**Comment by fflorent** [comment]
> Query parameters are generally (always?) lowercase, ideally one word, maybe with underscores. I suggest calling the param hidden, and updating /columns as well before merging https://github.com/gristlabs/grist-help/pull/256.

Seems reasonable. The GET verb already means that we fetch resources, and thus `include` in `includeHidden` is redundant.

**Comment by fflorent** [comment]
@alexmojaki I anticipated a bit and applied your suggestion :)

**Review by alexmojaki — APPROVED** [review]

**Pull request merged**